src/Nethermind/Nethermind.Optimism/OptimismBlockProcessor.cs

```diff
@@ -51,13 +55,29 @@ public OptimismBlockProcessor(
             executionRequestsProcessor)
     {
         ArgumentNullException.ThrowIfNull(stateProvider);
+        _opSpecHelper = opSpecHelper;
         _contractRewriter = contractRewriter;
+        _costHelper = costHelper;
         ReceiptsTracer = new OptimismBlockReceiptTracer(opSpecHelper, stateProvider);
     }
 
     protected override TxReceipt[] ProcessBlock(Block block, IBlockTracer blockTracer, ProcessingOptions options, IReleaseSpec spec, CancellationToken token)
     {
         _contractRewriter?.RewriteContract(block.Header, _stateProvider);
-        return base.ProcessBlock(block, blockTracer, options, spec, token);
+        TxReceipt[] receipts = base.ProcessBlock(block, blockTracer, options, spec, token);
+
+        if (_opSpecHelper.IsJovian(block.Header))
+        {
+            // TODO: handle overflow differently or calculate footprint as long?
```

Comment:
To avoid overflow do all computations in `UInt256`, and then cast to `long` and check for overflow